cumulusci/core/keychain/encrypted_file_project_keychain.py

```diff
@@ -160,12 +105,300 @@ def unset_default_org(self):
+        # The following steps occur in a _very_ particular order
+        self._create_default_service_files()
+        self._create_services_dir_structure()
+        self._migrate_services()
+        self._load_service_files()
+
+    def _load_service_files(self, constructor=None) -> None:
+        """
+        Load configured services onto the keychain.
+        This method recursively goes through all subdirectories
+        in ~/.cumulusci/services looking for .service files to load.
+        """
+        services_dir = Path(f"{self.global_config_dir}/services")
+        for item in services_dir.glob("**/*"):
+            if item.suffix == ".service":
+                with open(item) as f:
+                    config = f.read()
+                if "services" not in self.config:
+                    self.config["services"] = {}
+                name = item.name.replace(".service", "")
+                service_type = item.parent.name
+                if service_type not in self.config["services"]:
+                    self.config["services"][service_type] = {}
+                self.config["services"][service_type][name] = (
+                    constructor(config) if constructor else config
+                )
+
+    def _set_encrypted_service(self, service_type, alias, encrypted, project):
+        service_path = Path(
+            f"{self.global_config_dir}/services/{service_type}/{alias}.service"
+        )
+        with open(service_path, "wb") as f:
+            f.write(encrypted)
+
+    def _init_default_services(self) -> None:
+        """Init self._default_services on the keychain so that
+        calls to get_service() that do not pass an alias can
+        return the default service for the given type"""
+        global_default_services = Path(
+            f"{self.global_config_dir}/{DEFAULT_SERVICES_FILENAME}"
+        )
+        self._set_default_services_for_dir(global_default_services)
+
+        project_default_services = Path(
+            f"{self.project_local_dir}/{DEFAULT_SERVICES_FILENAME}"
+        )
+        # project defaults will overwrite global defaults
+        self._set_default_services_for_dir(project_default_services)
+
+    def _set_default_services_for_dir(self, dir_path: Path) -> None:
+        if dir_path.is_file():
+            with open(dir_path, "r") as f:
+                default_services = json.loads(f.read())
+
+            for s_type, alias in default_services.items():
+                self._default_services[s_type] = alias
+
+    def _save_default_services(self, project: bool = False) -> None:
+        """Write out the contents of self._default_services to the
+        DEFAULT_SERVICES.json file based on the provided scope"""
+        dir_path = (
+            Path(self.project_local_dir) if project else Path(self.global_config_dir)
+        )
+        with open(dir_path / DEFAULT_SERVICES_FILENAME, "w") as f:
+            f.write(json.dumps(self._default_services))
```

Comment:
Doesn't this mean that setting the default for one service type as a project-level default will turn the defaults for all other services into project-level defaults as well? I think when a default service is set at the project level we want to do a more targeted update of the project-level DEFAULT_SERVICES file to only add/update that one service in it.

Reply:
good catch